src/java/htsjdk/samtools/util/Interval.java

```diff
@@ -159,12 +159,17 @@ public int compareTo(final Interval that) {
 
     /** Equals method that agrees with {@link #compareTo(Interval)}. */
     public boolean equals(final Object other) {
-        if (!(other instanceof Interval)) return false;
-        else if (this == other) return true;
-        else {
-            Interval that = (Interval)other;
-            return (this.compareTo(that) == 0);
+        if (this == other) {
+            return true;
         }
+        if (other == null) {
+            return false;
+        }
+        if (this.getClass() != other.getClass()) { 
```

Comment:
@tfenne We primarily `Locatable` and gatk`s `SimpleInterval` so this wouldn't really effect our work.  It seems like a bad idea though.  Anyone inheriting from `Interval` is expecting this equality to work in one way and changing it now is going to cause major headaches.